module/os/linux/zfs/arc_os.c

```diff
@@ -204,8 +204,11 @@ arc_shrinker_count(struct shrinker *shrink, struct shrink_control *sc)
 	 *
 	 * See also the comment above zfs_arc_shrinker_limit.
 	 */
-	int64_t limit = zfs_arc_shrinker_limit != 0 ?
-	    zfs_arc_shrinker_limit : INT64_MAX;
+	int64_t limit = INT64_MAX;
+	if (current_is_kswapd()) {
+		limit = zfs_arc_shrinker_limit != 0 ?
+		    zfs_arc_shrinker_limit : INT64_MAX;
+	}
```

Comment:
minor suggestion instead:

```
if (current_is_kswapd() && zfs_arc_shrinker_limit != 0) {
    limit = zfs_arc_shrinker_limit;
}

```
... but not a big deal and just a matter of taste I suppose.

Reply:
I like it, thanks.